nifi-ngsi-bundle/nifi-ngsi-processors/src/main/java/org/apache/nifi/processors/ngsi/ngsi/backends/PostgreSQLBackend.java

```diff
@@ -167,30 +219,95 @@ else if ((NGSIConstants.ATTR_PER_COLUMN).equalsIgnoreCase(attrPersistence)) {
-                        valuesForColumns.put(encodedSubAttributeName, formatFieldForValueInsert(subAttribute.getAttrValue(), listOfFields.get(encodedSubAttributeName)));
-                  }
+            valuesForInsertList.add(valuesForInsert);
+        } else if ("ld".equals(ngsiVersion)) {
+            if (!isTemporalEntities) {
+                Map<String, String> valuesForColumns = new TreeMap<>();
+                int i = 0;
+                if (ckanCompatible) {
+                    valuesForColumns.put("_id", "'" + +i + "'");
+                }
+                ZonedDateTime creationDate = Instant.ofEpochMilli(creationTime).atZone(ZoneOffset.UTC);
+                valuesForColumns.put(NGSIConstants.RECV_TIME, "'" + DateTimeFormatter.ISO_INSTANT.format(creationDate) + "'");
+                valuesForColumns.put(NGSIConstants.ENTITY_ID, "'" + entity.getEntityId() + "'");
+                valuesForColumns.put(NGSIConstants.ENTITY_TYPE, "'" + entity.getEntityType() + "'");
+                for (AttributesLD attribute : entity.getEntityAttrsLD()) {
+                    String encodedAttributeName = encodeAttributeToColumnName(attribute.getAttrName(), attribute.getDatasetId(), datasetIdPrefixToTruncate);
+                    valuesForColumns.put(encodedAttributeName, formatFieldForValueInsert(attribute.getAttrValue(), listOfFields.get(encodedAttributeName)));
+                    if (attribute.isHasSubAttrs()) {
+                        for (AttributesLD subAttribute : attribute.getSubAttrs()) {
+                            String encodedSubAttributeName = encodeSubAttributeToColumnName(attribute.getAttrName(), attribute.getDatasetId(), subAttribute.getAttrName(), datasetIdPrefixToTruncate);
+                            valuesForColumns.put(encodedSubAttributeName, formatFieldForValueInsert(subAttribute.getAttrValue(), listOfFields.get(encodedSubAttributeName)));
+                        }
+                    }
+                }
+                valuesForInsertList.add("(" + String.join(",", valuesForColumns.values()) + ")");
+            } else{
+                Map<String, String> valuesForColumns = new TreeMap<>();
+                int i = 0;
+                if (ckanCompatible) {
+                    valuesForColumns.put("_id", "'" + +i + "'");
                 }
-            } //for
-            valuesForInsert = "(" + String.join(",", valuesForColumns.values()) + ")";
+                Map<String, List<AttributesLD>> attributesByObservedAt = entity.getEntityAttrsLD().stream().collect(Collectors.groupingBy(attrs -> attrs.observedAt));
+                List<String> timeStamps = attributesByObservedAt.keySet().stream().sorted().collect(Collectors.toList());
+                String oldestTimeStamp;
+
+                if(timeStamps.get(0).equals(""))oldestTimeStamp = timeStamps.get(1);
+                else oldestTimeStamp = timeStamps.get(0);
```

Comment:
why this?

Reply:
By adding the unobserved entities and sorting the timestamps. The first will be that of unobserved entities. so to take the oldest you have to take the second